annotation-processor/src/main/java/com/linecorp/armeria/server/annotation/processor/DocumentationProcessor.java

```diff
@@ -0,0 +1,186 @@
+            return false;
+        }
+        annotations.forEach(annotation -> processAnnotation(annotation, roundEnv));
+        propertiesMap.forEach((className, properties) -> {
+            try {
+                writeProperties(className, properties);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+        return false;
+    }
+
+    private Properties readProperties(String className) throws IOException {
+        if (propertiesMap.containsKey(className)) {
+            return propertiesMap.get(className);
+        }
+        final FileObject resource = processingEnv
+                .getFiler()
+                .getResource(StandardLocation.CLASS_OUTPUT,
+                             "",
+                             getFileName(className));
+        final Properties properties = new Properties();
+        if (resource.getLastModified() == 0L) {
+            // returns 0 if file does not exist
+            propertiesMap.put(className, properties);
+            return properties;
+        }
+        try (Reader reader = resource.openReader(false)) {
+            properties.load(reader);
+            return properties;
+        }
+    }
+
+    private void writeProperties(String className, Properties properties) throws IOException {
+        if (properties.size() == 0) {
+            return;
+        }
+        final FileObject resource = processingEnv.getFiler().createResource(
+                StandardLocation.CLASS_OUTPUT,
+                "",
+                getFileName(className));
+        try (Writer writer = resource.openWriter()) {
+            properties.store(writer, "Generated list of parameter description of REST interfaces.");
+        }
+    }
+
+    private void processAnnotation(TypeElement annotationElement, RoundEnvironment roundEnv) {
+        roundEnv.getElementsAnnotatedWith(annotationElement)
+                .stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                // Element is always ExecutableElement because it is a method.
+                .forEachOrdered(element -> {
+                    try {
+                        processMethod((ExecutableElement) element);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+    }
+
+    private void processMethod(ExecutableElement method) throws IOException {
+        final String className = ((TypeElement) method.getEnclosingElement()).getQualifiedName().toString();
+        final Properties properties = readProperties(className);
+        final String docComment = processingEnv.getElementUtils().getDocComment(method);
+        if (docComment == null || !docComment.contains("@param")) {
+            return;
+        }
+        final List<List<String>> lines = Arrays.stream(docComment.split("\\R"))
+                                               .filter(line -> !line.trim().isEmpty())
```

Comment:
How about using a `Splitter` so that we don't need this line?
```
private static final Splitter splitter = Splitter.on("\n").trimResults();
...
final List<List<String>> lines = Streams.stream(splitter.split(docComment)).map(...)
```